### PR TITLE
chore: Add Mailgun send email action

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Add GitHub Action to send email using `mailgun.js` npm module.
- Update nightly workflow to use the send email action.
- Add a failing integration test to trigger the email notification workflow (this test will be removed once we confirm the email notifications be working as expected).

### CODE REVIEW

File: .github/actions/send-email/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+function sendEmail() {
+  validate();
+
+  const mg = mailgun.client({
+    username: 'api',
+    key: apiKey,
+  });
+
+  let from = core.getInput('from');
+  const cc = core.getInput('cc');
```
Comment (lahirumaramba) [inline .github/actions/send-email/index.js:35]: We probably do not need `cc` filed. I am open to remove it and keep this action as simple as possible if we need.

File: .github/actions/send-email/index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    .then((resp) => {
+      core.setOutput('response', resp.message);
+      return;
+    })
+    .catch((err) => {
+      core.setFailed(err.message);
+    });
+}
+
+function validate() {
```
Comment (hiranya911) [inline .github/actions/send-email/index.js:62]: Validate all required parameters here. I'd expect more or less everything except cc is required.

File: .github/actions/send-email/index.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      "Recipient's email addresses must be a non-empty string."
+    );
+  }
+}
+
+function isNonEmptyString(value) {
+  return typeof value === 'string' && value !== '';
+}
+
+sendEmail();
```
Comment (hiranya911) [inline .github/actions/send-email/index.js:80]: I'd prefer something like:

```js
const config = loadConfig();
validate(config);
sendEmail(config);
```

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -66,3 +66,33 @@ jobs:
...
+      with:
+        api-key: ${{ secrets.OSS_BOT_MAILGUN_KEY }}
+        domain: ${{ secrets.OSS_BOT_MAILGUN_DOMAIN }}
+        from: 'GitHub <admin-github@${{ secrets.OSS_BOT_MAILGUN_DOMAIN }}>'
+        to: ${{ secrets.FIREBASE_ADMIN_GITHUB_EMAIL }}
+        subject: '[${{github.repository}}] Nightly build failed!'
+        html: >
+          <b>Nightly workflow failed on: ${{github.repository}}</b>
+          <br /><br />Checkout the 
+          <a href="github.com/${{github.repository}}/actions/workflows/nightly.yml">failed workflow</a>.
```
Comment (hiranya911) [inline .github/workflows/nightly.yml:82]: Let's link to https://github.com/firebase/firebase-admin-node/actions. Both here and below.

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Looks pretty good. A few thoughts for improvement.

**Comment by lahirumaramba** [comment]
Thanks, @hiranya911!
Used a bit of a lazy approach to validate all the required fields. Kept `cc`, `text`, `html`, as optional. If `html` is provided and not `text` the API extracts the text content from the `html`. Let me know what you think.

File: .github/actions/send-email/index.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    cc: core.getInput('cc'),
+    subject: core.getInput('subject'),
+    text: core.getInput('text'),
+    html: core.getInput('html'),
+  }
+}
+
+function validate(config) {
+  for (param in config) {
+    if (['cc', 'text', 'html'].includes(param)) {
```
Comment (hiranya911) [inline .github/actions/send-email/index.js:37]:
```
const optionalFields = ['cc', 'text', 'html'];
```

File: .github/actions/send-email/index.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      continue;
+    }
+    validateParameter(config[param], `'${param}'`);
+  }
+}
+
+function validateParameter(value, name) {
```
Comment (hiranya911) [inline .github/actions/send-email/index.js:44]: validateRequiredParameter

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Just a couple of readability nits. And remember to update the binary every time you push a code change.

**Comment by lahirumaramba** [comment]
Thanks @hiranya911! Updated the binary.

**Review by hiranya911 — APPROVED** [review]
LGTM 👍

**Comment by decodedfile** [comment]
Duplicate of #

**Pull request merged**